mkosi/__init__.py

```diff
@@ -1694,6 +1694,20 @@ def configure_dracut(state: MkosiState, cached: bool) -> None:
             '[[ $(modinfo -k "$kernel" -F filename efivarfs 2>/dev/null) == /* ]] && add_drivers+=" efivarfs "\n'
         )
 
+    # Modules may be required for device mapper targets
+    dm_modules = []
+    if state.config.encrypt:
+        dm_modules.append("dm-crypt")
+    if state.config.verity:
+        dm_modules.append("dm-verity")
+    if dm_modules:
+        dm_modules.append("dm-mod")
+
+        conf_content = 'add_dracutmodules+=" dm "\n'
+        for dm_module in dm_modules:
+            conf_content += f'[[ $(modinfo -k "$kernel" -F filename {dm_module} 2>/dev/null) == /* ]] && add_drivers+=" {dm_module} "\n'
```

Comment:
Isn't that something dracut does under the hood?

Reply:
To rephrase this: apparently not, otherwise you wouldn't have added this. Nevertheless, is there a reason the dracut `dm` module doesn't pull in these things as dependencies? I don't really think we want to reimplement that, since we could only do that badly and it would be brittle.